### PR TITLE
Update code to reduce redundancy and simplify readability

### LINKED ISSUE #29381: [Issue] Update code to reduce redundancy and simplify readability
This issue is automatically created based on existing pull request: magento/magento2#28608: Update code to reduce redundancy and simplify readability

---------
Simplified code complexity by optimizing logic, redundant variable assignments, over-usage of returns within a function.  Eliminated needless and redundant variable assignment.

### Preconditions: (*)

Simplifying logic and readability randomly

### Steps to reproduce (*)
1. Modified functions are simpler yet do not change any logic within `lib/internal/Magento/Framework/App/Request/Http.php`

**Actual Result:**

**Expected Result:**

### Questions or comments

No.

### Contribution checklist (*)
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [X] All new or changed code is covered with unit/integration tests (if applicable)
 - [X] All automated tests passed successfully (all builds are green)

### PR DESCRIPTION
**Pull request by df2k2**
Simplified code complexity by optimizing logic, redundant variable assignments, over-usage of returns within a function.  Eliminated needless and redundant variable assignment.

### Description (*)

Simplifying logic and readability randomly

### Manual testing scenarios (*)
1. Modified functions are simpler yet do not change any logic within `lib/internal/Magento/Framework/App/Request/Http.php`

### Questions or comments

No.

### Contribution checklist (*)
 - [X] Pull request has a meaningful description of its purpose
 - [X] All commits are accompanied by meaningful commit messages
 - [X] All new or changed code is covered with unit/integration tests (if applicable)
 - [X] All automated tests passed successfully (all builds are green)

### Resolved issues:
1. [x] resolves magento/magento2#29381: Update code to reduce redundancy and simplify readability

### CODE REVIEW

File: lib/internal/Magento/Framework/App/Request/Http.php [hunk 1]
```diff
@@ -292,13 +286,11 @@ public function initForward()
      * If passed name will be null whole state array will be returned.
      *
      * @param string $name
-     * @return array|string|null
+     * @return mixed|null
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Request/Http.php]: You should never use `mixed`

File: lib/internal/Magento/Framework/App/Request/Http.php [hunk 2]
```diff
@@ -228,8 +223,7 @@ public function getFrontName()
     public function setRouteName($route)
     {
         $this->route = $route;
-        $module = $this->routeConfig->getRouteFrontName($route);
-        if ($module) {
+        if ($module = $this->routeConfig->getRouteFrontName($route)) {
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Request/Http.php]: Assignment in `if` should not be used.

File: lib/internal/Magento/Framework/App/Request/Http.php [hunk 3]
```diff
@@ -292,13 +286,11 @@ public function initForward()
...
      * @param string $name
-     * @return array|string|null
+     * @return mixed|null
      */
     public function getBeforeForwardInfo($name = null)
     {
-        if ($name === null) {
-            return $this->beforeForwardInfo;
-        } elseif (isset($this->beforeForwardInfo[$name])) {
+        if ($name !== null && isset($this->beforeForwardInfo[$name])) {
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Request/Http.php]: As far as I see, you removed returning `$this->beforeForwardInfo`

File: lib/internal/Magento/Framework/App/Request/Http.php [hunk 4]
```diff
@@ -425,11 +413,8 @@ public function __sleep()
...
         if ($this->isSafeMethod === null) {
-            if (isset($_SERVER['REQUEST_METHOD']) && (in_array($_SERVER['REQUEST_METHOD'], $this->safeRequestTypes))) {
-                $this->isSafeMethod = true;
-            } else {
-                $this->isSafeMethod = false;
-            }
+            $this->isSafeMethod = isset($_SERVER['REQUEST_METHOD'])
+                && (in_array($_SERVER['REQUEST_METHOD'], $this->safeRequestTypes));
         }
         return $this->isSafeMethod;
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Request/Http.php]: I don't understand the usage of lazy loading there. HTTP request is stateful, so I'm not going to request clearing that.

File: lib/internal/Magento/Framework/App/Request/Http.php [hunk 5]
```diff
@@ -425,11 +413,8 @@ public function __sleep()
     public function isSafeMethod()
     {
         if ($this->isSafeMethod === null) {
-            if (isset($_SERVER['REQUEST_METHOD']) && (in_array($_SERVER['REQUEST_METHOD'], $this->safeRequestTypes))) {
-                $this->isSafeMethod = true;
-            } else {
-                $this->isSafeMethod = false;
-            }
+            $this->isSafeMethod = isset($_SERVER['REQUEST_METHOD'])
+                && (in_array($_SERVER['REQUEST_METHOD'], $this->safeRequestTypes));
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Request/Http.php]:
```suggestion
            $this->isSafeMethod = in_array($_SERVER['REQUEST_METHOD'] ?? null, $this->safeRequestTypes, true);
```

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Thank you for your contribution. Please review my Code Review comments.

File: lib/internal/Magento/Framework/App/Request/Http.php (hunk 2 above)
Comment (df2k2) [reply lib/internal/Magento/Framework/App/Request/Http.php]: Thanks for this.  I will change my habit from this feedback.

File: lib/internal/Magento/Framework/App/Request/Http.php (hunk 4 above)
Comment (df2k2) [reply lib/internal/Magento/Framework/App/Request/Http.php]:
```
if (isset($_SERVER['REQUEST_METHOD']) && (in_array($_SERVER['REQUEST_METHOD'], $this->safeRequestTypes))) {
  $this->isSafeMethod = true;
```

Is it safe to read the code above as `if (true) { $this->isSafeMethod = true; }` ?

and `} else { $this->isSafeMethod = false; }`

We can't just update this to be $this->isSafeMethod = (bool) result of isset($_SERVER['REQUEST_METHOD']) && (in_array($_SERVER['REQUEST_METHOD'], $this->safeRequestTypes)) ?

I appreciate the feedback.

Comment (lbajsarowicz) [reply lib/internal/Magento/Framework/App/Request/Http.php]: Yes, that was actually my point with "Lazy Load" part, as this should immediately return the `bool` value of this `in_array`

Comment (df2k2) [reply lib/internal/Magento/Framework/App/Request/Http.php]: Ok, I understand now.  Thank you

File: lib/internal/Magento/Framework/App/Request/Http.php (hunk 1 above)
Comment (df2k2) [reply lib/internal/Magento/Framework/App/Request/Http.php]: Thanks.  Note taken.

File: lib/internal/Magento/Framework/App/Request/Http.php (hunk 3 above)
Comment (df2k2) [reply lib/internal/Magento/Framework/App/Request/Http.php]: Good catch.  I've updated.

File: lib/internal/Magento/Framework/App/Request/Http.php (hunk 4 above)
Comment (df2k2) [reply lib/internal/Magento/Framework/App/Request/Http.php]: Updated back to previous.

**Review by df2k2 — COMMENTED** [review]
Updated

**Comment by lbajsarowicz** [comment]
@magento run all tests
Good afternoon @df2k2 , I'm running tests to make sure that everything works fine with your PR. If there are no issues, I am going to approve your changes and soon, all the changes should be processed by Core Team.

**Comment by lbajsarowicz** [comment]
@magento run Functional Tests CE

Looks like the failing test is flaky one. Trying to re-run it and in the meantime, I'm accepting the PR.

**Review by lbajsarowicz — APPROVED** [review]
:heavy_check_mark: Thank you for your contribution!

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7680](https://jira.corp.magento.com/browse/ENGCOM-7680) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Delta** [comment]
:heavy_check_mark: QA passed
Was checked general cases: place order as guest, register and sign in customer, place order as registered customer,add products to with list and compare list, works as before

**Comment by engcom-Echo** [comment]
@magento run all tests

File: lib/internal/Magento/Framework/App/Request/Http.php [hunk 6]
```diff
@@ -199,13 +199,8 @@ public function isDirectAccessFrontendName($code)
...
-        $path = parent::getBasePath();
-        if (empty($path)) {
-            $path = '/';
-        } else {
-            $path = str_replace('\\', '/', $path);
-        }
-        return $path;
+        return empty(parent::getBasePath()) ? '/'
+            : str_replace('\\', '/', parent::getBasePath());
     }
```
Comment (engcom-Echo) [inline lib/internal/Magento/Framework/App/Request/Http.php:204]: If 'parent: getBasePath()' is not empty, we will call 'parent::getBasePath()' again.
I think it is better to use the $ path variable.

File: lib/internal/Magento/Framework/App/Request/Http.php [hunk 7]
```diff
@@ -311,13 +305,9 @@ public function getBeforeForwardInfo($name = null)
...
-            return true;
-        }
-        if ($this->getParam('ajax') || $this->getParam('isAjax')) {
-            return true;
-        }
-        return false;
+        return $this->isXmlHttpRequest()
+            || $this->getParam('ajax', null)
+            || $this->getParam('isAjax', null);
     }
```
Comment (engcom-Echo) [inline lib/internal/Magento/Framework/App/Request/Http.php:311]: $this->getParam('ajax', null), second parameter is not required

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run Functional Tests EE

**Comment by engcom-Echo** [comment]
@magento run all tests

File: lib/internal/Magento/Framework/App/Request/Http.php [hunk 8]
```diff
@@ -311,13 +305,9 @@ public function getBeforeForwardInfo($name = null)
...
-        if ($this->isXmlHttpRequest()) {
-            return true;
-        }
-        if ($this->getParam('ajax') || $this->getParam('isAjax')) {
-            return true;
-        }
-        return false;
+        return $this->isXmlHttpRequest()
+            || $this->getParam('ajax', null)
+            || $this->getParam('isAjax', null);
```
Comment (lbajsarowicz) [inline lib/internal/Magento/Framework/App/Request/Http.php]: The default value is `null`, you do not need to provide it explicitly

`\Magento\Framework\App\Request\Http\Proxy::getParam($key, $default = null)`

**Review by lbajsarowicz — CHANGES_REQUESTED** [review]
Please review my comment regarding default values for `getParam`

**Comment by engcom-Echo** [comment]
@magento run all tests

**Review by lbajsarowicz — APPROVED** [review]
✔️ Thanks for your input.

**Comment by magento-engcom-team** [comment]
Hi @lbajsarowicz, thank you for the review.
[ENGCOM-7680](https://jira.corp.magento.com/browse/ENGCOM-7680) has been created to process this Pull Request
:eight_spoked_asterisk: @lbajsarowicz, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Alfa** [comment]
@lbajsarowicz Could you put an appropriate label for test coverage?

**Comment by lbajsarowicz** [comment]
@engcom-Alfa This class is one of the most-covered ones, that is the result of the dependencies relying on this class.

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed 
Was checked general cases ( create orders as guest and customer, register new customer, wishlist, and compare products) and everything works as expected.

**Comment by engcom-Alfa** [comment]
@magento create issue

**Pull request merged**